src/Drivers/Imagick/Core.php

```diff
@@ -98,10 +97,6 @@ public function empty(): CollectionInterface
      */
     public function slice(int $offset, ?int $length = null): CollectionInterface
     {
-        if ($offset >= $this->count()) {
```

Comment:
`array_slice([], 2, 1)` works and return `[]`.

I propose to allow this in order to avoid throwing exception in slice method (or you'll need to add `@throws RuntimeException` on every other method using slice).